src/framework/mpas_stream_list.F

```diff
@@ -118,15 +118,17 @@ subroutine MPAS_stream_list_insert(list, stream, ierr) !{{{
             list % head => stream
         else
             node => list % head
-            do while (associated(node % next))
+            do while (associated(node))
                 if (node % name == stream % name) then
                     if (present(ierr)) ierr = MPAS_STREAM_LIST_DUPLICATE
-                    LIST_ERROR_WRITE('Found duplicate item '//trim(stream % name)//' in list.')
```

Comment:
I don't think we need to remove this line, as the `LIST_ERROR_WRITE` macro is defined as a commented-out call to `mpas_log_write`.